src/Generator/Changelog/Changesets/ActionParam.php

```diff
@@ -0,0 +1,71 @@
+<?php
+namespace Mill\Generator\Changelog\Changesets;
+
+use Mill\Generator\Changelog\Changeset;
+
+class ActionParam extends Changeset
+{
+    /**
+     * @var array
+     */
+    protected $templates = [
+        'plural' => [
+            'added' => 'The following parameters have been added to {method} on {uri}:',
+            'removed' => 'The following parameters have been removed to {method} on {uri}:'
+        ],
+        'singular' => [
+            'added' => 'A {parameter} request parameter was added to {method} on {uri}.',
+            'removed' => 'The {parameter} request parameter has been removed from {method} requests on {uri}.'
+        ]
+    ];
+
+    /**
+     * @inheritdoc
+     */
+    public function compileAddedOrRemovedChangeset($definition, array $changes = [])
+    {
+        if (count($changes) === 1) {
+            $change = array_shift($changes);
+            $template = $this->templates['singular'][$definition];
+            return $this->renderText($template, $change);
+        }
+
+        $methods = [];
+        foreach ($changes as $change) {
+            $methods[$change['method']][] = $this->renderText('{parameter}', $change);
+        }
+
+        $entry = [];
+        foreach ($methods as $method => $params) {
+            if (count($params) > 1) {
+                $template = $this->templates['plural'][$definition];
+                $entry[] = [
+                    $this->renderText($template, [
+                        'method' => $method,
+                        'uri' => $changes[0]['uri']
+                    ]),
+                    $params
+                ];
+
+                continue;
+            }
+
+            $template = $this->templates['singular'][$definition];
+            $entry[] = $this->renderText($template, [
+                'parameter' => rtrim(ltrim(array_shift($params), '`'), '`'),
```

Comment:
That line of code was actually from when before I had built the Markdown format generator, and it was currently untested. Slightly rewrote the code above it to only templatize `$params` if we need to, and added some comments explaining this to futureself.